runtime/src/iree/hal/drivers/hip/rccl_status_util.h

```diff
@@ -0,0 +1,68 @@
+// Copyright 2024 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#ifndef IREE_HAL_DRIVERS_HIP_NCCL_STATUS_UTIL_H_
+#define IREE_HAL_DRIVERS_HIP_NCCL_STATUS_UTIL_H_
+
+#include <stdint.h>
+
+#include "iree/base/api.h"
+#include "iree/hal/drivers/hip/rccl_dynamic_symbols.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif  // __cplusplus
+
+// Converts a ncclResult_t to an iree_status_t.
+//
+// Usage:
+//   iree_status_t status = IREE_NCCL_RESULT_TO_STATUS(nccl_symbols,
+//                                                     ncclDoThing(...));
+#define IREE_NCCL_RESULT_TO_STATUS(syms, expr, ...) \
+  iree_hal_hip_nccl_result_to_status((syms), ((syms)->expr), __FILE__, __LINE__)
```

Comment:
Won't these conflict with the CUDA versions like https://github.com/iree-org/iree/blob/c1cfbfc48f0abb07ff2cfd6335eeb541097acda2/runtime/src/iree/hal/drivers/cuda/nccl_status_util.h#L19-L26 ? Maybe these should use `IREE_RCCL_` or they should be shared between CUDA and HIP (if the impls are the same, to save on binary size)